### PR TITLE
Add deprecation warning

### PR DESCRIPTION
**Pull request by caass**
Closes https://github.com/cloudflare/wrangler/issues/2261
Screen Shot 2022-06-22 at 7 25 59 AM

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -20,6 +22,22 @@ fn main() -> Result<()> {
     }
     env_logger::init();
 
+    StdErr::deprecation_warning(&format!(
+        r#"{} {} {}
+
+You are using an outdated version of Wrangler (v{}).
+Please upgrade to v2 ({}).
```
Comment (petebacondarwin) [inline src/main.rs:29]:
```suggestion
Please upgrade to the latest version ({}).
```

**Review by sidharthachatterjee — APPROVED** [review]
Looks good! This will be helpful in pushing folks to wrangler2

**Review by threepointone — CHANGES_REQUESTED** [review]
Blocking so we can remove "Deprecation warning" and make it less scary.

File: src/main.rs (hunk 1 above)
Comment (rozenmd) [reply src/main.rs:29]: The problem is that major versions aren't sacred, and we could make a wrangler v3 if we needed to introduce a breaking change.

So "the latest version" saves us from having to update it several times in the future.

File: src/main.rs [hunk 2]
```diff
@@ -20,6 +22,22 @@ fn main() -> Result<()> {
...
+Detailed migration instructions can be found here:
+{}
+"#,
+        emoji::WARN,
+        styles::warning("DEPRECATION WARNING"),
```
Comment (JacobMGEvans) [inline src/main.rs:34]: Rather than "Deprecation" 
```suggestion
        styles::warning("UPGRADE AVAILABLE"),
```

File: src/terminal/message.rs [hunk 3]
```diff
@@ -94,12 +94,22 @@ impl Message for StdErr {
...
+            .border_style(BorderStyle::Round)
+            .border_color(BorderColor::Cyan)
+            .margin(1)
+            .build();
+        eprint!("{}", billboard.as_str(msg));
     }
 
-    fn deprecation_warning(_msg: &str) {
-        panic!("Can't display billboard warning to stderr.")
+    fn deprecation_warning(msg: &str) {
```
Comment (JacobMGEvans) [inline src/terminal/message.rs:106]:
```suggestion
    fn upgrade_message(msg: &str) {
```

**Comment by caass** [comment]
If memory serves, we had decided to incorporate this instead into the normal self-update check, so that's what I've done here.

**Review by petebacondarwin — CHANGES_REQUESTED** [review]

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**